### DISCUSSION TITLE
Declare css variable with Tailwind

### DISCUSSION THREAD

**Discussion by Sulaymon-dev** [tailwindlabs/tailwindcss#6145] [Ideas]
Can I declare CSS variables with Tailwind like var-[--gap-x:2rem] md:var-[--gap-x:3rem] xl:var-[--gap-x:4rem] ???

**Comment by adamwathan** [comment]
I like this idea! Need to think through the exact syntax more but I bet we can do something like this. My gut is to honestly not even make it variable-specific, just let you set any arbitrary property/value combo:

```html
<div class="[--gap-x:2rem] md:[--gap-x:3rem] xl:[--gap-x:4rem]"></div>
<div class="[paint-order:markers] md:[paint-order:normal]"></div>
```

**Comment by Sulaymon-dev** [comment]
That's right. This syntax is much better. Will this feature be added in version 3 of the Tailwind?

**Comment by mariohamann** [comment]
Hey!

I've had a similar idea and have built a plugin around the concept of setting and getting variables via classes:
- Demos: https://mariohamann.de/tailwindcss-var
- Plugin: https://github.com/mariohamann/tailwindcss-var

Some Playground examples:
- Basic color change: https://play.tailwindcss.com/8IoJCK0r1t
- Hierarchy and subvalues: https://play.tailwindcss.com/fQPslGpLCw
- Sizes: https://play.tailwindcss.com/ojWbNZGdMY
- Arbitrary values: https://play.tailwindcss.com/ETCMSUlWiv

Especially working with colors in a component-based setup is absolutely fantastic with this concept (already tried it out in different settings).

I would love to see something like that in core. ❤

BTW: I've experimented with different syntax and the syntax I've used in the plugin feels super natural to me. Would love to hear your opinion on that.
